crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2469,6 +2484,69 @@ impl std::fmt::Display for DynamicType {
     }
 }
 
+bitflags! {
+    /// Type qualifiers that appear in an annotation expression.
+    #[derive(Copy, Clone, Debug, Eq, PartialEq)]
+    pub(crate) struct TypeQualifiers: u8 {
+        /// `typing.ClassVar`
+        const CLASS_VAR = 1 << 0;
+        /// `typing.Final`
+        const FINAL     = 1 << 1;
+    }
+}
+
+impl TypeQualifiers {
+    pub(crate) fn is_class_var(self) -> bool {
```

Comment:
strictly this method seems to be querying whether `ClassVar` is _one_ of the qualifiers for the variable rather than whether it _is the sole_ qualifier for the variable, so maybe this might be a better name?

```suggestion
    pub(crate) fn includes_class_var(self) -> bool {
```

I think it's actually invalid to combine `ClassVar` with any other type qualifiers currently (`ClassVar[Final[int]]` is not permitted). But this might change in the future, and we probably want to be able to recover gracefully and infer a good type even if the user does do something illegal with their qualifiers. So using a bitflag rather than an enum seems like a good option here.

Reply:
Hm.. for the sake of the argument, imagine that `ClassVar` and `Final` would be allowed, then I think it would still be okay, terminology wise, to say that a particular variable "*is* a classvar and also *is* final"?

I agree that "includes" makes more sense when thinking from the perspective of the set of type qualifiers (which is what this struct represents).

I have now moved this function to `SymbolAndQualifiers` which changes the code that uses it to:
```rs
let instance_member = instance.class.instance_member(self.db(), attr);
if instance_member.is_class_var() {
    // emit diagnostic
}
```

What do you think?

Reply:
Ah, that reads much better if it's a method on `SymbolAndQualifiers`. This is resolved now in that case -- thanks!